contracts/modules/TransferManager/VolumeRestrictionTransferManager.sol

```diff
@@ -0,0 +1,241 @@
+            }
+            return Result.INVALID;
+        }
+        return Result.NA;
+    }
+    
+    // lets the owner / admin create a volume restriction lockup.
+    // takes a userAddress and lock up params, and creates a lockup for the user.  See the LockUp struct def for info on what each parameter actually is
+    function addLockUp(address userAddress, uint lockUpPeriodSeconds, uint releaseFrequencySeconds, uint startTime, uint totalAmount) public withPerm(ADMIN) {
+        
+        // if a startTime of 0 is passed in, then start now.
+        if (startTime == 0) {
+            startTime = now;
+        }
+        
+        // Get an array of the user's lockups
+        LockUp[] storage userLockUps = lockUps[userAddress];
+        
+        // create a new lock up and push it into the array of the user's lock ups
+        LockUp memory newLockUp;
+        newLockUp.lockUpPeriodSeconds = lockUpPeriodSeconds;
+        newLockUp.releaseFrequencySeconds = releaseFrequencySeconds;
+        newLockUp.startTime = startTime;
+        newLockUp.totalAmount = totalAmount;
+        userLockUps.push(newLockUp);
+
+        emit LogModifyLockUp(
+            userAddress,
+            lockUpPeriodSeconds,
+            releaseFrequencySeconds,
+            startTime,
+            totalAmount,
+            LockUpOperationType.Add
+        );
+    }
+
+    // same as addLockup, but takes an array for each parameter
+    function addLockUpMulti(address[] userAddresses, uint[] lockUpPeriodsSeconds, uint[] releaseFrequenciesSeconds, uint[] startTimes, uint[] totalAmounts) public withPerm(ADMIN) {
+        
+        // make sure input params are sane
+        require(
+            userAddresses.length == lockUpPeriodsSeconds.length && 
+            userAddresses.length == releaseFrequenciesSeconds.length &&
+            userAddresses.length == startTimes.length &&
+            userAddresses.length == totalAmounts.length,
+            "Input array length mis-match"
+        );
+
+        for (uint i = 0; i < userAddresses.length; i++) {
+            addLockUp(userAddresses[i], lockUpPeriodsSeconds[i], releaseFrequenciesSeconds[i], startTimes[i], totalAmounts[i]);
+        }
+
+    }
+    
+    // remove a user's lock up
+    function removeLockUp(address userAddress, uint lockUpIndex) public withPerm(ADMIN) {
+        LockUp[] storage userLockUps = lockUps[userAddress];
+        LockUp memory toRemove = userLockUps[lockUpIndex];
+
+        emit LogModifyLockUp(
+            userAddress,
+            toRemove.lockUpPeriodSeconds,
+            toRemove.releaseFrequencySeconds,
+            toRemove.startTime,
+            toRemove.totalAmount,
+            LockUpOperationType.Remove
+        );
+
+        // move the last element in the array into the index that is desired to be removed.  
+        userLockUps[lockUpIndex] = userLockUps[userLockUps.length - 1];
+        // delete the last element
+        userLockUps.length--;
```

Comment:
Try to add a check 

         if (lockUpIndex < userLockUps.length - 1) {
             userLockUps[lockUpIndex] = userLockUps[userLockUps.length - 1];
           } 
          userLockUps.length--;

Using this we will save un-necessary state variable re-writing i.e save gas.